zyte_parsers/utils.py

```diff
@@ -80,7 +84,18 @@ def extract_link(a_node: SelectorOrElement, base_url: str) -> Optional[str]:
     except ValueError:
         link = None
 
-    return link
+    if not link or not force_safe:
+        return link
+
+    try:
+        safe_link = safe_url_string(link)
+    except ValueError:
+        return None
+
+    # add scheme (https) when missing schema and no base url
+    safe_link = add_https_to_url(safe_link)
```

Comment:
Super minor: I think "no base url" part of the comment does not quite apply, as this can have effect even if base URL is supplied. Also if we take that away we'd just repeat the function name so maybe we could remove the comment completely.